### ISSUE TITLE
Proposal: Provide a PlaceholderForeground API for the NumberBox control

### ISSUE THREAD

**Issue by Felix-Dev** [microsoft/microsoft-ui-xaml#2811]
# Proposal: Provide a PlaceholderForeground API for the NumberBox control

## Summary
NumberBox has a [Placeholder API](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.numberbox.placeholdertext?view=winui-2.4#Microsoft_UI_Xaml_Controls_NumberBox_PlaceholderText) just like the TextBox and ComboBox to set a Placeholder. What NumberBox currently lacks compared to them is a [PlaceholderForeground API](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.textbox.placeholderforeground?view=winrt-19041#Windows_UI_Xaml_Controls_TextBox_PlaceholderForeground) which can be used to set the foreground color of the Placeholder. Developers coming from the TextBox/CombBox using their PlaceholderForeground API can be assumed to expect the existence of such an API for the NumberBox as well.

In addition, lightweight-styling the placeholder foreground for the NumberBox currently does not work. The core issue behind this is mentioned here: https://github.com/microsoft/microsoft-ui-xaml/issues/2774#issuecomment-651848564 Internally, the NumberBox uses a TextBox which controls the placeholder and as the issue mentioned above affects the TextBox, so does it affect the NumberBox.

## Rationale
Provide a consistent API for controls with matching functionality to meet expectations of developers.

## Scope
| Capability | Priority |
| :---------- | :------- |
| Add a PlaceholderForeground API to the NumberBox control. | Must |
| Make the new PlaceholderForeground API work on all supported Windows version by WinUI. | Should | 

## Important Notes
I have some potentially bad news here. My current test implementation of this new API template binds the the NumberBox.PlaceholderForeground API to the NumberBox's internal TextBox.PlaceholderForeground API. Unfortunately, the TextBox.PlaceholderForeground API has only been added in Windows 10 1709 to the XAML platform so it is not available for us to use on Windows 10 1703 - the current earliest Windows version WinUI 2 supports. I did experiment with trying to work around this API limitation on 1703, but with limited success only. The TextBox uses an internal TextBlock to set the placeholder and while I managed to grab that TextBlock of the TextBox used by the NumberBox, I was not successful to set the TextBlock.Foreground color in all required visual states though (such as NumberBox receiving focus, or the mouse pointer entering the NumberBox's input field). As such, it currently appears to me that it will be quite challenging, if not nearly impossible, to match the PlaceholderForeground API behavior on 1709 and above on 1703.

How much is that a concern here? Does that rule out adding this API to the NumberBox until WinUI 3?

## API Proposal
```cs
public class NumberBox
{
    public static DependencyProperty PlaceholderForegroundProperty { get; }

    public Windows.UI.Xaml.Media.Brush PlaceholderForeground { get; set; }
}
```

### API example
```xml
<NumberBox
    PlaceholderText="Placeholder"
    PlaceholderForeground="Blue"/>
```
Resulting look:
![image](https://user-images.githubusercontent.com/1398851/86303362-f20fb800-bc0b-11ea-9578-7ae76726b6f4.png)

## Additional Notes
I have the NumberBox.PlaceholderForeground API ready in a local branch of mine and could work on a PR and API spec - if approved by the team.

**Comment by StephenLPeters** [comment]
Personally I think it is fine for this to No-Op on RS2. The text box doesn't have this customization on that platform so I think its fine for numberbox to not have it either. I think using conditional xaml this would be easy to achieve.

**Comment by StephenLPeters** [comment]
@SavoySchuler @MikeHillberg and @teaP FYI

**Comment by Felix-Dev** [comment]
@StephenLPeters 
> I think using conditional xaml this would be easy to achieve.

Yep, that's exactly how I would do it. Should I go ahead and create a PR and a possible API spec (seems like @SavoySchuler is currently busy)?

**Comment by MikeHillberg** [comment]
I'd like to work on that, combined with InputScope and the TextAlignment discussion.

**Comment by Felix-Dev** [comment]
@MikeHillberg Alright, I'll defer to you then.
